Add builder + add examples + add tests

u9g opened the pull request:
this has breaking functionality !!!

changes: 
1. an object is exported instead of a single function
2. expanded tests
3. added a messagebuilder
4. moved & added examples

extremeheat commented on builder.js:
```diff
@@ -0,0 +1,36 @@
+  setTranslate (val) { this.translate = val; return this }
+}
+
+class MessageComponent extends baseMessage {
```
What's the difference between MessageComponent and Message? They seem the same

u9g replied:
none

extremeheat commented on builder.js:
```diff
@@ -0,0 +1,36 @@
+class baseMessage {
```
should be uppercase, BaseMessage

extremeheat commented on builder.js:
```diff
@@ -0,0 +1,36 @@
+  }
+}
+
+class Message extends baseMessage {
```
add in a way to pass an existing json object to Message that gets merged into this

u9g commented:
Changes made by this PR:
1. moved the old index.js to ChatMessage.js to allow for a loader
2. made a builder class in builder.js - this class allows building messages using the message class, which was previously only possible by directly writing json
3. added convertColorCodes method which converts color codes to the json form, which can be then used to make a ChatMessage Object
4. Heavily expanded on tests
5. Expanded examples
6. (WIP) added docs for new functions

rom1504 commented:
Can you explain why any alternative without breaking change is worse ?

u9g commented:
> Can you explain why any alternative without breaking change is worse ?

what alternative are you suggesting?

Karang commented on ChatMessage.js:
```diff
@@ -0,0 +1,327 @@
+
+function loader (mcVersion) {
+  const mcData = require('minecraft-data')(mcVersion)
+  defaultLang = mcData.language
```
Don't define `defaultLang` and `ChatMessage` in global, but put them inside the loader function. Otherwise the code will break as soon as you try instantiating multiple versions. (we had a similar issue in anvil)

Karang reviewed (commented):
Minor comment, otherwise LGTM

extremeheat commented on MessageBuilder.js:
```diff
@@ -0,0 +1,98 @@
+      const color = supportedColors[text.shift()]
+      const newObj = new MessageBuilder()
+      if (color === 'reset') {
+        newObj.setObfuscated(false)
```
It's a new object, is all this necessary? Or does undefined have a different meaning? If so, you can add a reset() method

u9g replied:
undefined means something different since we are just trying to break inheritance of any of these properties. The code isn't used again so I don't know if its worth it to make it a function.

rom1504 commented:
what's the status on this @u9g ?

u9g commented:
This is "done" for text but a lot needs to be added like passing a p-item as a hoveritem, I dont really need it for what I was doing in fs, so i never did that part

builder-247 commented:
Can this be merged? I'd like to use  MessageBuilder in https://github.com/PrismarineJS/mineflayer/pull/1984

u9g commented:
> Can this be merged? I'd like to use MessageBuilder in [PrismarineJS/mineflayer#1984](https://github.com/PrismarineJS/mineflayer/pull/1984)

it has a lot of bad things in it, like that it is breaking, imo this should be refactored into a nonbreaking pr
